src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py

```diff
@@ -476,94 +476,71 @@ def get_chart_mapping_schema(
 
         with open(values_schema, "r", encoding="utf-8") as f:
             data = json.load(f)
-            schema_data = data["properties"]
+            schema_data = data
 
         try:
-            final_schema = self.find_deploy_params(values_data, schema_data, {})
+            deploy_params_dict = self.traverse_dict(values_data, DEPLOYMENT_PARAMETER_MAPPING_REGEX)
+            new_schema = self.search_schema(deploy_params_dict, schema_data)
         except KeyError as e:
             raise InvalidTemplateError(
-                f"ERROR: Your schema and values for the helm package '{helm_package.name}' do not match. \
+                f"ERROR: There is a problem with your schema or values for the helm package '{helm_package.name}'. \
```

Comment:
Can we add something a bit more explicit here? Like what is the key we failed to find the schema for?

Reply:
we have relaxed the rules about matching schemas here so i think a more generic answer is ok for now, people are unlikely to hit this. I could just remove the try except completely tbh